mantidproject/mantid
# NormaliseToMonitor detector scan improvements

Some minor tidy-ups to the algorithm.

1. The divide step for detector scan workspaces is now done in an OpenMP loop
1. The check for a monitor ID being in a workspace is only done if there are less than 10,000 histograms present
1. There is a redundant warning for single count workspaces about integration ranges that has been removed
1. Having a separate monitor workspace is not supported properly for detector scan workspaces - now a proper error is thrown
1. Progress tracking has been added

**To test:**

Use the attached file for D2B - [d20_det_scan.zip].(https://github.com/mantidproject/mantid/files/1526808/d20_det_scan.zip)


1. Check running `NormaliseToMonitor` is quicker on this branch than master
1. Check the algorithm dialogue is quicker to open for this workspace than on master
1. Check no "Integration range" warning is shown
1. Use `ExtractMonitors` to create a separate detector and monitor workspace. Try running `NormaliseToMonitor` giving the separate monitor workspace - an error should be thrown.
1. Check progress tracking is sensible for running on the D20 workspace

Fixes #21025.

**Release Notes** 
*Does not need to be in the release notes.* This is a tidy-up for a previous ticket to support scanning workspaces in `NormaliseToMonitor`.


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
